### PR TITLE
Minor refactoring/cleanup of tests/test_sockets.py. NFC

### PR DESCRIPTION
**Pull request by sbc100**
I have a followup that paramaterizes these tests so that there is
just one harness used per test, and this precursor is mostly to
make that change simpler and easier to review.

Them main change here is to have the harness `__enter__` function
return self it can be used in a `with .. as ..:` expression

### CODE REVIEW

**Comment by sbc100** [comment]
Best reviewed with ignore whitespace

File: tests/test_sockets.py [hunk 1]
```diff
@@ -53,7 +53,9 @@ def __enter__(self):
     # NOTE empty filename support is a hack to support
     # the current test_enet
     if self.filename:
-      run_process([CLANG_CC, test_file(self.filename), '-o', 'server', '-DSOCKK=%d' % self.target_port] + clang_native.get_clang_native_args() + self.args, env=clang_native.get_clang_native_env())
+      cmd = [CLANG_CC, test_file(self.filename), '-o', 'server', '-DSOCKK=%d' % self.target_port] + clang_native.get_clang_native_args() + self.args
+      print(cmd)
```
Comment (dschuff) [inline tests/test_sockets.py:57]: is it intentional to leave this print in place? (fine with me if so, just checking)

**Review by dschuff — APPROVED** [review]

File: tests/test_sockets.py (hunk 1 above)
Comment (sbc100) [reply tests/test_sockets.py:57]: Yup, we want to see the command name.  But I just remembered we have self.run_process for this.. so I'll switch to that.

Comment (sbc100) [reply tests/test_sockets.py:57]: Actually this code is not withing a test suite.. so I can't get to self.run_process from here.

**Pull request merged**